services/subject_recommendation/subject_recommendation_system.py

```diff
@@ -0,0 +1,201 @@
+import pandas as pd
+import numpy as np
+from sentence_transformers import SentenceTransformer, util
+from sklearn.cluster import KMeans
+from collections import Counter
+from queue import PriorityQueue
+
+class SubjectRecommendationSystem:
+    def __init__(self, list_of_boards, subjects_queryset):
+        self.list_of_boards = list_of_boards
+        # Filter for both ssc and hsc accordingly. Take in both
+        # the curriculum will be passed in the API call. Filter the dataframe accordingly.
+        self.required_columns = ["name", "curriculum__abbreviation", "education_level"]
+        # self.df = pd.DataFrame(list(subjects_queryset.values_list(*self.required_columns, flat=True)), columns=["name"])
+
+        self.data = list(subjects_queryset.values_list(*self.required_columns))
+        self.df = pd.DataFrame(self.data, columns=self.required_columns)
+
+        #print(self.df.head())
+        self.subjects_list = self.df['name'].tolist()
+        self.model = SentenceTransformer('multi-qa-MiniLM-L6-cos-v1')
+        self.subjects_vectors = self.encode_subjects()
+        
+        self.init_recomm_for_each_board = {}
+        self.all_boards_streams_files = {}
+
+        for board in self.list_of_boards:
```

Comment:
Please add comments defining, what is happening in this step

Reply:
The code has been significantly changed in the latest commit. I have added comments and docstrings wherever necessary.